tests/Unit/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -10,9 +11,91 @@
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Requires.hpp"
 #include "Utilities/Tuple.hpp"
+#include "tests/Unit/TestingFramework.hpp"
 
 namespace TestHelpers {
 namespace VectorImpl {
+
+namespace detail {
+// `check_vectors` checks equality of vectors and related data types for tests.
+//
+// Comparisons between any pairs of the following are supported:
+// vector types
+// arithmetic types
+// arrays of vectors
+// arrays of arithmetic types
+
+// between two vectors
+template <typename T1, typename T2,
+          Requires<cpp17::is_arithmetic_v<typename T1::ElementType> and
+                   cpp17::is_arithmetic_v<typename T2::ElementType>> = nullptr>
+inline void check_vectors(const T1& t1, const T2& t2) noexcept {
```

Comment:
Is it possible to just do `const VectorImpl<T, VT>&, const VectorImpl<T, VT>&` because of the implicit conversion from blaze::DenseVector to VectorImpl? If so that would seem like a cleaner way to implement all the "switching" than the `Requires`